java/code/src/com/redhat/rhn/common/db/datasource/xml/System_queries.xml

```diff
@@ -2261,6 +2261,32 @@ SELECT channel_arch_id
   </query>
 </mode>
 
+<mode name="xmlrpc_empty_profiles" class="com.redhat.rhn.frontend.dto.BootstrapSystemOverview">
+    <query params="org_id, user_id">
+        SELECT
+        1 AS selectable,
+        0 AS locked,
+        rhnServer.id,
+        rhnServer.name,
+        TO_CHAR(checkin, 'YYYY-MM-DD HH24:MI:SS') AS last_checkin,
+        rhnCPU.nrcpu as cpu_count,
+        rhnCPU.mhz as cpu_mhz,
+        rhnRAM.ram as ram,
+        (SELECT COUNT(*) FROM rhnDevice WHERE rhnDevice.server_id = rhnServer.id AND class = 'HD') AS disk_count
+        FROM rhnServer
+        INNER JOIN suseMinionInfo on rhnServer.id = suseMinionInfo.server_id
+        LEFT JOIN rhnServerEntitlementView ON rhnServerEntitlementView.server_id = rhnServer.id
+        LEFT JOIN rhnUserServerPerms ON rhnServer.id = rhnUserServerPerms.server_id
```

Comment:
Are you sure that a "left join" of the permissions table is correct? If the permissions are allowed to not exist it might become useless.

Or do have bootstrap systems no permissions entries here?

Reply:
It would work even in the suggested case - this "optional case" wouldn't go through because of these conditions in `WHERE`:
* `rhnServerEntitlementView.label = 'bootstrap_entitled'`
* `rhnUserServerPerms.user_id = :user_id`

But using `INNER JOIN` would be more correct in this case.